src/encoded/cart_view.py

```diff
@@ -1,14 +1,26 @@
 from pyramid.view import view_config
 from collections import OrderedDict
+from snovault import COLLECTIONS
+from pyramid.httpexceptions import HTTPBadRequest
 
+CART_USER_MAX = 30  # Maximum number of non-deleted carts allowed per user
 
 def includeme(config):
     # config.scan()
-    # config.add_route('carts', '/carts{slash:/?}')
     config.add_route('cart-view', '/cart-view{slash:/?}')
+    config.add_route('cart-manager', '/cart-manager{slash:/?}')
     config.scan(__name__)
 
 
+def get_cart_objects_by_user(request, userid):
+    request.datastore = 'database'
+    return [
+        request.embed(request.resource_path(v), '@@object')
+        for k, v in request.registry[COLLECTIONS]['cart'].items()
+        if v.properties['submitted_by'] == userid and (v.properties['status'] != 'deleted' or 'group.admin' in request.effective_principals)
```

Comment:
This ones a little long too, can you expand with parens?

Reply:
Changed with another change.